program/src/instructions/compressed/buy_spl.rs

```diff
@@ -109,6 +125,7 @@ pub struct BuySpl<'info> {
 
 impl<'info> Validate<'info> for BuySpl<'info> {
     fn validate(&self) -> Result<()> {
+        // TODO: is there a reason these validation checks are done differently between ixs? Eg this account is checked via anchor in legacy ix
```

Comment:
Backwards compatibility hack to support both old fee accounts and new sharded fee vaults.